backend/api/v1/models.py

```diff
@@ -125,7 +143,61 @@ class Skill(db.Model):
     slug = db.Column(IdentifierType, primary_key=True)
     name = db.Column(db.String(50), nullable=False)
 
-    content = db.relationship("Content", back_populates="skills", secondary="content_skill_association")
+    content = db.relationship(
+        "Content", back_populates="skills", secondary="content_skill_association"
+    )
 
-    def __repr__(self):
+    def __repr__(self):  # pragma: no cover
         return f"<Skill(name='{self.name}')>"
+
+
+class Submission(db.Model):
+    __tablename__ = "submissions"
+
+    slug = db.Column(IdentifierType, primary_key=True)
+    content_slug = db.Column(
+        IdentifierType, db.ForeignKey("contents.slug"), nullable=False
+    )
+    date = db.Column(db.DateTime, nullable=False)
+    user = db.Column(db.String, nullable=False)
+    external_identifier = db.Column(db.String, nullable=True)
+
+    notebooks = db.relationship("Notebook", back_populates="submission")
+    content = db.relationship("Content")
+
+    @property
+    def score(self):
+        return sum(n.score for n in self.notebooks)
+
+    @property
+    def max_score(self):
+        return sum(n.max_score for n in self.notebooks)
+
+    @property
+    def graded(self):
+        return (
+            all(notebook.graded for notebook in self.notebooks)
+            if self.notebooks
+            else False
+        )
+
+    def __repr__(self):  # pragma: no cover
+        return f"<Submission(user='{self.user}', maxScore={self.max_score})>"
+
+
+class Notebook(db.Model):
+    __tablename__ = "notebooks"
+
+    slug = db.Column(IdentifierType, primary_key=True)
+    name = db.Column(db.String(50), nullable=False)
+    submission_slug = db.Column(
+        IdentifierType, db.ForeignKey("submissions.slug"), nullable=False
+    )
+    score = db.Column(db.Float, nullable=False, default=0)
+    max_score = db.Column(db.Float, nullable=False, default=0)
+    graded = db.Column(db.Boolean, nullable=True, default=False)
```

Comment:
Why should this be `nullable`? I'd think it should be `False` by default and never be null.

Reply:
That is because `sqlalchemy` has some serious problems with boolean columns and default values.... 
If I set `default=False`, it actually thinks it should not set a default (instead of using the default set to False). Fortunately, None or False have more or less the same meaning in python (in most of the cases), so this has the same effect. 

Googling a bit, it might also be possible to set a server default. I will try that.

Reply:
Ok, I managed to fix this. Thanks for insisting on that!